pilot/pkg/serviceregistry/kube/mcs/discoverability.go

```diff
@@ -0,0 +1,303 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package mcs
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"istio.io/istio/pilot/pkg/model"
+	kubesr "istio.io/istio/pilot/pkg/serviceregistry/kube"
+	"istio.io/istio/pkg/config/host"
+	"istio.io/istio/pkg/config/schema/gvk"
+	"istio.io/istio/pkg/queue"
+	"istio.io/pkg/log"
+	"istio.io/pkg/monitoring"
+)
+
+var (
+	scope = log.RegisterScope("mcs", "logging scope for MCS functionality", 0)
+
+	empty = struct{}{}
+
+	eventTag   = monitoring.MustCreateLabel("event")
+	clusterTag = monitoring.MustCreateLabel("cluster")
+
+	serviceExportEvents = monitoring.NewSum(
+		"service_export_events",
```

Comment:
it feels a bit odd to have a metric explicitly for service_export. Can we just make it a label on `type` on an existing metric here? We already have the CRD events and service registry events one I think

Reply:
if not - I think the prometheus best practice is to suffix with _total

Reply:
Also - mcs_export_events may be more readable. We  also have 'service export' as a core istio feature ( and all other CRs have export)

Reply:
This comment  no longer applies.